### PR TITLE
[CRD] Add field vpcPath in VPCNetworkConfiguration

### PR DESCRIPTION
**Pull request by wenyingd**

### CODE REVIEW

File: pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go [hunk 1]
```diff
@@ -46,6 +46,9 @@ type VPCNetworkConfigurationSpec struct {
 	// +kubebuilder:validation:MaxLength=8
 	// +optional
 	ShortID string `json:"shortID,omitempty"`
+	// NSXTVPC is the NSX-T VPC the Namespace associated with.
+	// +optional
+	NSXTVPC string `json:"nsxtVPC,omitempty"`
```
Comment (dantingl) [inline pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go:51]: how about VPCPath?

File: pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go [hunk 2]
```diff
@@ -46,6 +46,9 @@ type VPCNetworkConfigurationSpec struct {
 	// +kubebuilder:validation:MaxLength=8
 	// +optional
 	ShortID string `json:"shortID,omitempty"`
+	// NSXTVPC is the NSX-T VPC the Namespace associated with.
```
Comment (dantingl) [inline pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go:49]: NSX-T -> NSX

Comment (wenyingd) [reply pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go:49]: I saw that `NSXTProject` also uses "NSX-T Project" in the comment. If here I modify as "NSX", do you mean NSXProject also needs update?

File: pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go (hunk 1 above)
Comment (wenyingd) [reply pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go:51]: Updated.

File: pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go [hunk 3]
```diff
@@ -46,6 +46,9 @@ type VPCNetworkConfigurationSpec struct {
 	// +kubebuilder:validation:MaxLength=8
 	// +optional
 	ShortID string `json:"shortID,omitempty"`
+	// Policy path of the NSX VPC the Namespace associated with.
+	// +optional
+	VPCPath string `json:"vpcPath,omitempty"`
 }
```
Comment (zhengxiexie) [inline pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go:52]: You can only modify this file, and then `make manifests` will generate the new yaml files, `make generated` will copy it to `nsx.vmware.com` dir and generate other clients by code-generator.

File: pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go [hunk 4]
```diff
@@ -46,6 +46,9 @@ type VPCNetworkConfigurationSpec struct {
 	// +kubebuilder:validation:MaxLength=8
 	// +optional
 	ShortID string `json:"shortID,omitempty"`
+	// Policy path of the NSX VPC the Namespace associated with.
```
Comment (lxiaopei) [inline pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go:49]: Could we change to "Policy path" to "NSX path"? Users of NSX operator CRs are confused for "policy path".

File: pkg/apis/nsx.vmware.com/v1alpha1/vpcnetworkconfiguration_types.go [hunk 5]
```diff
@@ -46,6 +46,9 @@ type VPCNetworkConfigurationSpec struct {
 	// +kubebuilder:validation:MaxLength=8
 	// +optional
 	ShortID string `json:"shortID,omitempty"`
+	// Policy path of the NSX VPC the Namespace associated with.
+	// +optional
+	VPCPath string `json:"vpcPath,omitempty"`
```
Comment (lxiaopei) [inline pkg/apis/nsx.vmware.com/v1alpha1/vpcnetworkconfiguration_types.go:51]: VpcPath?

**Comment by zhengxiexie** [comment]
BTW, could you update build/yaml/samples/nsx_v1alpha1_vpcnetworkconfigurations.yml ?

File: build/yaml/crd/nsx.vmware.com_vpcnetworkconfigurations.yaml [hunk 6]
```diff
@@ -94,6 +94,9 @@ spec:
                   context in logs. Less than or equal to 8 characters.
                 maxLength: 8
                 type: string
+              vpcPath:
+                description: Policy path of the NSX VPC the Namespace associated with.
```
Comment (lxiaopei) [inline build/yaml/crd/nsx.vmware.com_vpcnetworkconfigurations.yaml:98]: In the description, we also need to make it clear that if vpcPath is provided, then other fields except defaultSubnetSize， podSubnetAccessMode do not take effect. 
could you also add a sample under build/yaml/samples?

File: pkg/apis/nsx.vmware.com/v1alpha1/vpcnetworkconfiguration_types.go (hunk 5 above)
Comment (wenyingd) [reply pkg/apis/nsx.vmware.com/v1alpha1/vpcnetworkconfiguration_types.go:51]: VPC is the abbreviation of Virtual Private Cloud, we always use VPC but not Vpc to present this meaning, just like NSX. So I don't think we shall use camel style for it.

File: build/yaml/crd/nsx.vmware.com_vpcnetworkconfigurations.yaml (hunk 6 above)
Comment (wenyingd) [reply build/yaml/crd/nsx.vmware.com_vpcnetworkconfigurations.yaml:98]: Updated.

File: pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go (hunk 4 above)
Comment (wenyingd) [reply pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go]: updated.

**Comment by wenyingd** [comment]
Having offline sync with @lxiaopei, adding a separate VPCNetworkConfiguration CR sample focusing on pre-created VPC scenario makes it less confusion, updated accordingly.

File: pkg/apis/nsx.vmware.com/v1alpha1/vpcnetworkconfiguration_types.go (hunk 5 above)
Comment (dantingl) [reply pkg/apis/nsx.vmware.com/v1alpha1/vpcnetworkconfiguration_types.go:53]: VPCPath +1

Comment (lxiaopei) [reply pkg/apis/nsx.vmware.com/v1alpha1/vpcnetworkconfiguration_types.go:53]: As discussed in google chat， Could we change to ”VPC“？

Comment (wenyingd) [reply pkg/apis/nsx.vmware.com/v1alpha1/vpcnetworkconfiguration_types.go:53]: Sure, will update.

File: build/yaml/samples/nsx_v1alpha1_vpcnetworkconfigurations.yaml [hunk 7]
```diff
@@ -13,3 +13,14 @@ spec:
...
   defaultSubnetAccessMode: Private
+---
+# Sample to create VPCNetworkConfiguration CR using a pre-created NSX VPC.
+apiVersion: nsx.vmware.com/v1alpha1
+kind: VPCNetworkConfiguration
+metadata:
+  name: vpc-network-config-with-pre-created-vpc
+spec:
+  nsxtProject: proj-2
+  vpc: /orgs/default/projects/proj-1/vpcs/vpc-1
```
Comment (lxiaopei) [inline build/yaml/samples/nsx_v1alpha1_vpcnetworkconfigurations.yaml:24]: we do not nsxtProject param since the we have the whole vpc path.

**Review by lxiaopei — COMMENTED** [review]
a minor comment, others LGTM.

File: pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go (hunk 3 above)
Comment (wenyingd) [reply pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go:54]: The latest version has performed the required make operations.

File: build/yaml/samples/nsx_v1alpha1_vpcnetworkconfigurations.yaml (hunk 7 above)
Comment (wenyingd) [reply build/yaml/samples/nsx_v1alpha1_vpcnetworkconfigurations.yaml:23]: Removed accordingly.

**Review by lxiaopei — APPROVED** [review]

**Review by dantingl — APPROVED** [review]

**Pull request merged**